docs/docker.md

```diff
@@ -0,0 +1,48 @@
+# Run a Function in a Docker Container
+
+To run your function in a container, create a `Dockerfile` with the following contents:
+
+```Dockerfile
+# Use the official Node.js 10 image.
+# https://hub.docker.com/_/node
+FROM node:10
+# Create and change to the app directory.
+WORKDIR /usr/src/app
+# Copy application dependency manifests to the container image.
+# A wildcard is used to ensure both package.json AND package-lock.json are copied.
+# Copying this separately prevents re-running npm install on every code change.
+COPY package.json package*.json ./
+# Install production dependencies.
+RUN npm install --only=production
+# Copy local code to the container image.
+COPY . .
+# Run the web service on container startup.
+CMD [ "npm", "start" ]
+```
+
+Start the container locally by running `docker build` and `docker run`:
+
+```sh
+docker build -t helloworld . && docker run -p 8080:8080 -it helloworld
+```
+
+Send requests to this function using `curl` from another terminal window:
+
+```sh
+curl localhost:8080
+# Output: Hello, World
+```
+
+## Deploy a Container
+
+You can deploy your containerized function to Cloud Run or any Knative-based environment by following the [Cloud Run quickstart](https://cloud.google.com/run/docs/quickstarts/build-and-deploy).
+
+Use the `docker` and `gcloud` CLIs to build and deploy a container to Cloud Run, replacing the project id `$GOOGLE_CLOUD_PROJECT` and image name `helloworld`:
+
+```sh
+docker build -t gcr.io/$GOOGLE_CLOUD_PROJECT/helloworld .
+docker push gcr.io/$GOOGLE_CLOUD_PROJECT/helloworld
```

Comment:
Note docker push requires some extra steps to get docker authenticating with GCP. In fact, if you use Cloud Build you don't need to get docker working locally at all.

Here are some [Cloud Run-specific docs on build options](https://cloud.google.com/run/docs/building/containers)

Reply:
How do we want to change these steps? Do we want to link the build options?

Reply:
Based on your needs, pick Cloud Build or Docker and include mostly bare steps here, linking to that page for more context/options?

Reply:
E.g., to keep the current instructions, add `gcloud auth configure-docker`.

Reply:
I've added the configure docker command.